pkgs/top-level/all-packages.nix

```
@@ -16957,6 +16957,8 @@ with pkgs;
 
   arrow-cpp = callPackage ../development/libraries/arrow-cpp {};
 
+  arrow-cpp-glib = callPackage ../development/libraries/arrow-cpp-glib {};
```

```suggestion
  arrow-glib = callPackage ../development/libraries/arrow-glib {};
```
https://arrow.apache.org/docs/c_glib/

---

Or `arrow-c-glib`